src/enum.cr

```diff
@@ -509,7 +509,24 @@ abstract struct Enum
   # If multiple members match the same normalized string, the first one is returned.
   def self.parse?(string : String) : self?
     {% begin %}
-      case string.gsub('-', '_').camelcase.downcase
+      # FIXME: There is no `StringLiteral#bytesize` or any other adequate means
+      # to figure out how much space we actually need. Maybe some regex could
+      # work. For now just play it safe.
+      # FIXME: We might want to establish some upper limit in case a member name
+      # is exorbitantly long.
+
+      # The following is an optimized normalization. It is equivalent to
+      # `string.gsub('-', '_').camelcase.downcase` but does not allocate.
+      buffer = uninitialized UInt8[{{ @type.constants.map(&.size).sort.last * 4 + 1 }}]
+      max_size = {{ @type.constants.map(&.size).sort.last }}
```

Comment:
I don't claim to understand macros very well, so maybe this is a dumb question, but...
Couldn't buffer depend on max_size here? The sorting and mapping is done twice at the moment.

```crystal
max_size = {{ @type.constants.map(&.size).sort.last }}
buffer = uninitialized UInt8[max_size * 4 + 1]
```

Reply:
I updated the code with this. There's no runtime performance benefit, but it does look cleaner.